packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -558,8 +565,8 @@ function isRefIdentifier(id: Identifier, parent: _Node, parentStack: _Node[]) {
 const isStaticProperty = (node: _Node): node is Property =>
   node && node.type === 'Property' && !node.computed
 
-const isStaticPropertyKey = (node: _Node, parent: _Node) =>
-  isStaticProperty(parent) && parent.key === node
+const isStaticPropertyKey = (node: _Node, parent: _Node, prop: string) =>
+  isStaticProperty(parent) && prop === 'key' && parent.key === node
```

Comment:
In acorn, when a property is shorthand version (e.g. `const foo = { bar }`), `key` and `value` has a different reference. But in rollup v4, those have a same reference.
https://stackblitz.com/edit/node-duaz1w?file=index.js

So it wasn't possible to check whether the `node` is the key of the `parent` or it is the value of it.

Reply:
I was also stumbling across this. Considering you also stumbled across this, maybe I do a last-minute change here.

Reply:
I added a fix to rollup/rollup#5169

Reply:
@sapphi-red in this case could we remove the `prop` changes here?

Reply:
Removed the changes 👍